src/Time/Time.php

```diff
@@ -21,6 +21,63 @@
 
 class Time implements TimeContract
 {
+
+    /**
+     * Date format.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+    */
+    private $dateFormat;
+
+    /**
+     * Set the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return object
+     */
+     public function setDateFormat($format)
+     {
+        if ($this->validateDateFormat($fornat)) {
+            $this->dateFormat = $format;
+
+            return $this;
+        }
+
+        throw new \InvalidArgumentException("The format {$format} is not valid format", 500);
+        
+     }
+
+    /**
+     * Get the time format.
+     *
+     * @since 3.0.0
+     *
+     * @return string
+     */
+     public function getDateFormat()
+     {
+        return $this->dateFormat;        
+     }
+
+    /**
+     * Validate the time format.
+     *
+     * @param mixe $format Valiud Date/Time format.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+     */
+     public function setDateFormat($format)
```

Comment:
What kinds of types for `$format`? The `DateTime` class and `string`?

Reply:
Typo

Reply:
Yeah its Date Tine class

Reply:
@Lablnet, thanks.

But it seems that the `$format` type is mixed` inside the this method comment annotation.

The `mixed` type comment annotation means this `$format` can allow two or more types.

Reply:
yes i will fix this but how/what about the validation process

Reply:
Could you illustrate the `validation` approach?

Reply:
idk you mentioned in issue